### PR TITLE
Monte Carlo in C

### PR DESCRIPTION
**Pull request by OkarinRintaro**
Calculating pi and integral by method Monte Carlo in C

Changes:

Created integral_montecarlo.c
Created pi_montecarlo.c

### CODE REVIEW

File: code/numerical_analysis/monte_carlo/src/integral_montecarlo.c [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#include <stdlib.h>
+
+/*Part of Cosmos by OpenGenus Foundation*/
+
+double
+integral_monte_carlo(double a, double b, double point_count, double (*f)(double))
+{
+    double res, x, y;
+    y = 0;
+    for (int i = 0; i < point_count; ++i)
```
Comment (YashovardhanSiramdas) [inline code/numerical_analysis/monte_carlo/src/integral_montecarlo.c]: 1)Declare looping variables outside the loop for compatibility with older C versions. 
2)Opening brace for loop should be in the same line(line 11)

File: code/numerical_analysis/monte_carlo/src/integral_montecarlo.c [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+/*Part of Cosmos by OpenGenus Foundation*/
+
+double
+integral_monte_carlo(double a, double b, double point_count, double (*f)(double))
+{
+    double res, x, y;
+    y = 0;
```
Comment (arnavb) [inline code/numerical_analysis/monte_carlo/src/integral_montecarlo.c]: No need for this to be on its own line (unless the style guide says so.)

File: code/numerical_analysis/monte_carlo/src/integral_montecarlo.c [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+{
+    double a, b, point_count, res;
+    printf("Enter interval: \n");
+    printf("a: \n");
+    scanf("%lf", &a);
+    printf("b: \n");
+    scanf("%lf", &b);
+    printf("Enter point count: ");
+    scanf("%lf", &point_count);
+    res = integral_monte_carlo(a, b, point_count, ({double f(double x){return x*x + x;} f;}));
```
Comment (arnavb) [inline code/numerical_analysis/monte_carlo/src/integral_montecarlo.c]: This line could be formatted better.

File: code/numerical_analysis/monte_carlo/src/integral_montecarlo.c [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    printf("Result =  %lf \n", res);
```
Comment (arnavb) [inline code/numerical_analysis/monte_carlo/src/integral_montecarlo.c]: the variable res is pretty unnecessary.

File: code/numerical_analysis/monte_carlo/src/pi_montecarlo.c [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+#include <stdlib.h>
+
+/*Part of Cosmos by OpenGenus Foundation*/
+
+double
+pi_monte_carlo(double point_count)
+{
+    double circle_points, pi, x, y;
+    circle_points = 0;
+    for (int i = 0; i < point_count; ++i)
```
Comment (arnavb) [inline code/numerical_analysis/monte_carlo/src/pi_montecarlo.c]: Same as what @YashovardhanSiramdas said.

File: code/numerical_analysis/monte_carlo/src/pi_montecarlo.c [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        x = (double)random() / RAND_MAX;
+        y = (double)random() / RAND_MAX;
+        if (x * x + y * y < 1)
+            circle_points++;
```
Comment (arnavb) [inline code/numerical_analysis/monte_carlo/src/pi_montecarlo.c]: Pre increment saves copy.

File: code/numerical_analysis/monte_carlo/src/pi_montecarlo.c [hunk 7]
```diff
@@ -0,0 +1,31 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+/*Part of Cosmos by OpenGenus Foundation*/
+
+double
+pi_monte_carlo(double point_count)
+{
+    double circle_points, pi, x, y;
+    circle_points = 0;
```
Comment (arnavb) [inline code/numerical_analysis/monte_carlo/src/pi_montecarlo.c]: Same as above.

File: code/numerical_analysis/monte_carlo/src/integral_montecarlo.c (hunk 2 above)
Comment (YashovardhanSiramdas) [reply code/numerical_analysis/monte_carlo/src/integral_montecarlo.c]: @arnavb Style Guide says that opening brace should be in the same line,only exception being functions.

File: code/numerical_analysis/monte_carlo/src/integral_montecarlo.c (hunk 4 above)
Comment (OkarinRintaro) [reply code/numerical_analysis/monte_carlo/src/integral_montecarlo.c]: @arnavb What do you mean?

File: code/numerical_analysis/monte_carlo/src/integral_montecarlo.c (hunk 2 above)
Comment (arnavb) [reply code/numerical_analysis/monte_carlo/src/integral_montecarlo.c]: I'm talking about the y = 0 being on its own line.

File: code/numerical_analysis/monte_carlo/src/integral_montecarlo.c (hunk 4 above)
Comment (arnavb) [reply code/numerical_analysis/monte_carlo/src/integral_montecarlo.c]: You don't need the variable at all, instead just do:

```C
printf("Result =  %lf \n", integral_monte_carlo(a, b, point_count, ({double f(double x){return x*x + x;} f;})));
```
Just a recommendation.

**Comment by OkarinRintaro** [comment]
@arnavb , @YashovardhanSiramdas, Thank you for advices!

File: code/numerical_analysis/monte_carlo/src/pi_montecarlo.c [hunk 8]
```diff
@@ -6,14 +6,13 @@
...
-    for (int i = 0; i < point_count; ++i)
-    {
+    double pi, x, y, circle_points = 0;
+    int i;
+    for (i = 0; i < point_count; ++i){
         x = (double)random() / RAND_MAX;
         y = (double)random() / RAND_MAX;
         if (x * x + y * y < 1)
-            circle_points++;
+            circle_points += 1;
```
Comment (arnavb) [inline code/numerical_analysis/monte_carlo/src/pi_montecarlo.c]: Not what I meant. Do ++circle_points instead.

Comment (OkarinRintaro) [reply code/numerical_analysis/monte_carlo/src/pi_montecarlo.c]: Did it.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**